### ISSUE TITLE
[Discussion] Deprecation and Backward Compatibility

### ISSUE THREAD

**Issue by windkit** [leo-project/leofs#625]
## Description
I think it would be better to declare some features/options/... to be deprecated in next major version.
So we can drop the backward compatibility up to certain point. This could help to relieve the burden of taking care of old code.

## Suggestion
For example, we can reserve x.0.z to be backward compatibile checkpoint, any version from (x-1).y.z could update to that but not to newer. (With conversion command to migrate data/...). We can then drop the compatibility code after x.1.z

**Comment by mocchira** [comment]
Just following the majority convention http://semver.org/ sounds good to me.

**Comment by windkit** [comment]
@mocchira Agree, that's what I would like to refer to too.

**Comment by mocchira** [comment]
@windkit OK.
@yosukehara thought?
if no problem, we should state that somewhere in the document now we are working on.

**Comment by yosukehara** [comment]
@windkit @mocchira I agree with adhering to the guideline, and I/we will announce that on the documentation.

```
Given a version number MAJOR.MINOR.PATCH, increment the:

MAJOR version when you make incompatible API changes,
MINOR version when you add functionality in a backwards-compatible manner, and
PATCH version when you make backwards-compatible bug fixes.
Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
```
